### PR TITLE
Execute all cleanup steps, not just the last one

### LINKED ISSUE #8: submit with --ebs not cleaning up EBS volumes
When submitting a job with the `--ebs` option, the created EBS volume is not cleaned up from the EC2 instance, rather it stays attached to the EC2 instance for the duration of the instance lifetime.

I suspect this `trap` command:
https://github.com/base2genomics/batchit/blob/master/submit/submit.go#L116
overrides this `trap` command:
https://github.com/base2genomics/batchit/blob/master/submit/submit.go#L218

### PR DESCRIPTION
**Pull request by tomazberisa**
Resolves #8

### CODE REVIEW

File: submit/submit.go [hunk 1]
```diff
@@ -207,15 +208,15 @@ func Main() {
 			n = 2
 		}
 		if len(ebs) == 4 {
-			ebsCmd[0] = fmt.Sprintf("export vid=$(batchit ebsmount -n %d -m %s -s %s -v %s -t %s)", n, ebs[0], ebs[1], ebs[2], ebs[3])
+			ebsCmd[1] = fmt.Sprintf("export vid=$(batchit ebsmount -n %d -m %s -s %s -v %s -t %s)", n, ebs[0], ebs[1], ebs[2], ebs[3])
```
Comment (brentp) [inline submit/submit.go]: this should be now ebs 1 through 4. but better yet. don't prepend the function to the ebsCmd array, just add the new string to the arguments list in the sprintF before ebsCmd[0].

File: submit/submit.go [hunk 2]
```diff
@@ -178,8 +178,8 @@ func Main() {
 			return
 		}
 	}
-	var ebsCmd [4]string
-	ebsCmd[0] = `cleanup_volume() { echo "batchit: No volumes to clean up"; }`
+	cleanupDefault := `cleanup_volume() { echo "batchit: No volumes to clean up"; }`
```
Comment (tomazberisa) [inline submit/submit.go]: This look better?

Comment (brentp) [reply submit/submit.go]: yes. just make the default be the empty function with no echo.

File: submit/submit.go [hunk 3]
```diff
@@ -113,7 +113,7 @@ func getTmp(cli *cliargs) string {
 	tmp := fmt.Sprintf(`# thanks Hao
 export TMPDIR="$(mktemp -d -p %s)"
 cleanup() { echo "batchit: deleting temp dir ${TMPDIR}"; umount -l /tmp/; rm -rf ${TMPDIR}; }
-trap cleanup EXIT
+trap "cleanup_volume EXIT; cleanup;" EXIT
```
Comment (brentp) [inline submit/submit.go:116]: got an extra EXIT in this line?

Comment (tomazberisa) [reply submit/submit.go:116]: cleanup_volume() takes the signal as a parameter

Comment (brentp) [reply submit/submit.go:116]: got it. can you verify this works on your set-up and then i'll merge?

Comment (tomazberisa) [reply submit/submit.go:116]: confirmed, successfully ran latest code update with and without the `--ebs` option

**Comment by brentp** [comment]
thanks very much.

**Pull request merged**